packages/api/endpoints/providers.js

```diff
@@ -131,32 +133,39 @@ async function post(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function put({ params: { id }, body }, res) {
-  if (id !== body.id) {
+  const apiProvider = body;
+
+  if (id !== apiProvider.id) {
     return res.boom.badRequest(
       `Expected provider ID to be '${id}', but found '${body.id}' in payload`
     );
   }
 
-  const knex = await getKnexClient({ env: process.env });
+  const knex = await getKnexClient();
   const providerModel = new Provider();
-
-  const providerExists = await Promise.all([
-    providerModel.exists(id),
-    doesRecordExist({ name: id }, knex, tableNames.providers),
-  ]);
-
-  if (providerExists.filter((providerTest) => providerTest === true).length !== 2) {
+  let oldProvider;
+  try {
+    oldProvider = await providerModel.get({ id });
+  } catch (error) {
+    if (error.name !== 'RecordDoesNotExist') {
+      throw error;
+    }
     return res.boom.notFound(
-      `Provider with ID '${id}' not found in Dynamo and Postgres databases`
+      `Provider with ID '${id}' not found`
     );
   }
 
+  apiProvider.updatedAt = Date.now();
+  apiProvider.createdAt = oldProvider.createdAt;
+
   let record;
-  let createObject = await translateApiProviderToPostgresProvider(body);
-  createObject = nullifyUndefinedProviderValues(createObject);
+  const createObject = await translateApiProviderToPostgresProvider(apiProvider);
+
   await knex.transaction(async (trx) => {
-    await trx(tableNames.providers).where({ name: id }).update(createObject);
-    record = await providerModel.create(body);
+    await trx(tableNames.providers)
+      .where({ name: id })
+      .update(nullifyUndefinedProviderValues(createObject));
```

Comment:
An update in this way is consistent with existing create-as-update logic in Dynamo, however it may be worth considering if that's a correct/consistent behavior for the API.